src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Tasks/UseValueTaskCorrectlyTests.cs

```diff
@@ -1074,6 +1097,28 @@ public void StoreIntoFields()
             );
         }
 
+        [Fact]
+        public async Task Diagnostics_Discards()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(CSBoilerplate(@"
+                using System;
+                using System.Threading.Tasks;
+
+                class C
+                {
+                    public void Discards()
+                    {
+                        _ = Helpers.ReturnsValueTask();
+                        _ = Helpers.ReturnsValueTaskOfT<string>();
+                        _ = Helpers.ReturnsValueTaskOfInt();
```

Comment:
💡 Consider adding lines showing that calling `AsTask()` or `Preserve()` on each of these will not result in a diagnostic when assigned to a discard.